src/components/atoms/SupplierCoupons/index.tsx

```diff
@@ -1,62 +1,116 @@
 import React from "react";
 import Image, { StaticImageData } from "next/image";
 import CouponPrimary from "@/images/easytolive/icons/couponPrimary.svg";
+import CouponRed from "@/images/easytolive/icons/couponred.svg";
+import CouponGreen from "@/images/easytolive/icons/coupongreen.svg";
 import ShoppingCartGreen from "@/images/easytolive/icons/shopping_cart_green.svg";
 import ClockCircleRed from "@/images/easytolive/icons/clock_circleRed.svg";
 import classNames from "@/utils/classNames";
 
 interface SupplierCouponsProps {
+  title: string;
   discount: string;
-  unintsAmount: number;
-  expirateTime: number;
+  unintsAmount: number | string;
+  expirateTime: number | string;
   icon: string | StaticImageData;
+  status?: "active" | "used" | "expired" | string;
+  usageDate?: string;
 }
 
 const SupplierCoupons: React.FC<SupplierCouponsProps> = ({
+  title,
   discount,
   unintsAmount,
   expirateTime,
   icon,
+  usageDate,
+  status = "active",
 }) => {
   return (
-    <div className="bg-primary-main h-auto pl-4 rounded-full flex items-center gap-3 cursor-pointer">
-      <h2 className=" font-bold text-white text-xl">{discount}%</h2>
+    <div
+      className={`${
+        status == "used"
+          ? "bg-secondary-main"
+          : status == "expired"
+          ? "bg-generic-alertRed"
+          : "bg-primary-main"
+      } h-auto pl-5 rounded-full flex items-center gap-2 cursor-pointer`}
+    >
+      <h2
+        className={`font-semibold text-xl ${
+          status == "used" ? "text-primary-main" : "text-white"
+        }`}
+      >
+        {discount}%
+      </h2>
 
       <div
         className={classNames(
-          "rounded-full bg-white w-full py-3 gap-4 -m-[1px] hover:shadow-md",
+          "rounded-full bg-white w-full py-3 gap-2 -m-[1px] hover:shadow-md",
         )}
       >
         <div className="Rounded-full flex items-center justify-evenly w-full">
           <Image
             className="h-10 w-auto"
             alt="Coupons Image"
-            src={CouponPrimary}
+            src={status == "expired" ? CouponRed : CouponPrimary}
```

Comment:
u can compare  type and value, pls use ===
```suggestion
            src={status === "expired" ? CouponRed : CouponPrimary}
```